src/backend/app/tasks/tasks_schemas.py

```diff
@@ -135,3 +137,46 @@ class ReadTask(Task):
     """Task details plus updated task history."""
 
     task_history: Optional[List[TaskHistoryOut]] = None
+
+
+class TaskHistory(BaseModel):
+    """Task history details."""
+
+    model_config = ConfigDict(
+        from_attributes=True,
+    )
+
+    # Excluded
+    user: Any = Field(exclude=True)
+
+    task_id: int
+    action_text: str
+    action_date: datetime
+
+    @computed_field
```

Comment:
I moved all the logic to the pydantic model.

Note we can use `@computed_field` decorators to generated computed values.
We can also use this to generate values based on database relationships.
To do this we need the model_config `from_attributes=True`:
```python
    model_config = ConfigDict(
        from_attributes=True,
    )
```